### PR TITLE
feat[added authentication]

### PR DESCRIPTION
**Pull request by Radheshyam1234**
Added login and signUp page along with the authentication context and state context

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
+import { useNavigate } from "react-router-dom";
 import { Routes, Route } from "react-router-dom";
 import { useProductDataProvider } from "./Context/ProductDataContext/ProductDataProvider";
 import { getProductsFromDb } from "./utilities/backendRequest";
+import { useStateProvider } from "./Context/StateContext/state-provider";
+import { useAuthProvider } from "./Context/AuthContext/AuthProvider";
 
-import { Home, ProductListing } from "./Components";
+import { Home, ProductListing, Login, SignUp } from "./Components";
+import { getUserProfileFromDb } from "./utilities/backendRequest";
```
Comment (junaidshaikh-js) [inline src/App.js:10]: We can name this only `getUserProfile`

File: src/App.js [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
 
 import "./styles.css";
 
 export const App = () => {
   const { setProductList } = useProductDataProvider();
+  const { setUserProfile } = useAuthProvider();
+  const { userState, userDispatch } = useStateProvider();
```
Comment (junaidshaikh-js) [inline src/App.js:17]: There is no need to add `provider` at the end of this custom hooks, you can only name them as `useProduct()` `useAuth()` `useState()`

File: src/Components/Authentication/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { useAuthProvider } from "../../../Context/AuthContext/AuthProvider";
+export const Login = () => {
```
Comment (junaidshaikh-js) [inline src/Components/Authentication/Login/Login.js:5]: Add empty lines between two logical blocks 

Like here, between import and the component there should be an empty line indicating the separation of code.

File: src/Components/index.js [hunk 4]
```diff
@@ -2,3 +2,5 @@ export { Home } from "./Home";
 export { ProductListing } from "./Product-listing";
 export { LikeBtn } from "./LikeBtn/LikeBtn";
 export { AddToCartBtn } from "./AddToCartBtn/AddToCartBtn";
+export { SignUp } from "./Authentication";
+export { Login } from "./Authentication";
```
Comment (junaidshaikh-js) [inline src/Components/index.js]: Can be exported in single line

**Review by junaidshaikh-js — COMMENTED** [review]
Good Work! 

You have written clean code. We can reduce the extra naming of variables and some good formatting practices.

I have mentioned points in the comments.

File: src/App.js [hunk 5]
```diff
@@ -1,21 +1,41 @@
...
+
+  const token = localStorage.getItem("token");
+
+  useEffect(async () => {
+    if (token) {
+      getUserProfileFromDb(setUserProfile);
+      navigate("/");
+    }
+  }, [token]);
+
```
Comment (prachi-sahani) [inline src/App.js:32]: multiple useEffects?

File: src/App.js [hunk 6]
```diff
@@ -1,21 +1,41 @@
 import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
 import { Routes, Route } from "react-router-dom";
 import { useProductDataProvider } from "./Context/ProductDataContext/ProductDataProvider";
 import { getProductsFromDb } from "./utilities/backendRequest";
+import { useStateProvider } from "./Context/StateContext/state-provider";
```
Comment (prachi-sahani) [inline src/App.js:6]: useStateProvider can named better as it is ambiguous

**Review by prachi-sahani — COMMENTED** [review]
Apart from the mentioned points, the code looks clean and is good to merge

**Pull request merged**